src/Microsoft.AspNetCore.Razor.Language/DirectiveTokenEditHandler.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    internal class DirectiveTokenEditHandler : SpanEditHandler
+    {
+        public DirectiveTokenEditHandler(Func<string, IEnumerable<ISymbol>> tokenizer) : base(tokenizer)
+        {
+        }
+
+        protected override PartialParseResult CanAcceptChange(Span target, SourceChange change)
+        {
+            if (AcceptedCharacters == AcceptedCharacters.NonWhiteSpace)
+            {
+                var currentWhitespace = CountWhitespace(target.Content);
```

Comment:
Would it be better to check if there are **any** whitespace characters in the edited region before and after?

This counting business seems pretty fuzzy